kamel-core/src/commonMain/kotlin/io/kamel/core/fetcher/HttpFetcher.kt

```diff
@@ -29,15 +28,15 @@ internal class HttpFetcher(private val client: HttpClient) : Fetcher<Url> {
         data: Url,
         resourceConfig: ResourceConfig
     ): Flow<Resource<ByteReadChannel>> = channelFlow {
-        val response = client.request<HttpResponse> {
+        val response = client.request {
             onDownload { bytesSentTotal, contentLength ->
                 val progress = (bytesSentTotal.toFloat() / contentLength).coerceAtMost(1.0F)
                 send(Resource.Loading(progress))
             }
             takeFrom(resourceConfig.requestData)
             url(data)
         }
-        val bytes = response.receive<ByteReadChannel>()
+        val bytes = response.body<ByteReadChannel>()

```

Comment:
Tested this in Ktor 2.0.0, it needs to be
```suggestion
        val bytes = response.bodyAsChannel()
```